contract_compassion/views/contract_view.xml

```diff
@@ -58,7 +58,10 @@
                                options="{'create': false, 'm2o_dialog': false}"/>
                         <field name="group_id" domain="[('partner_id', '=', partner_id)]"
                                context="{'default_partner_id': partner_id}"/>
-                        <field name="parent_id" domain="[('partner_id', '=', partner_id), ('child_id', '!=', child_id)]"/>
+                        <field name="parent_id" domain="[('partner_id', '=', partner_id), ('child_id', '!=', child_id), ('sub_sponsorship_id', '=', False)]"/>
+                    </group>
+                    <group>
+                        <field name="sub_sponsorship_id"/>
```

Comment:
Don't put it in a separate group. You can add it just after field parent_id. Hide the field when the sponsorship is not finished by adding the attribute `states=terminated`